rasa/train.py

```diff
@@ -491,19 +491,31 @@ async def _train_core_with_validated_data(
         return _train_path
 
 
-def _core_model_for_finetuning(
+async def _core_model_for_finetuning(
     model_to_finetune: Text,
-    new_config: Optional[Dict] = None,
+    file_importer: TrainingDataImporter,
     finetuning_epoch_fraction: float = 1.0,
 ) -> Optional[Agent]:
     path_to_archive = model.get_model_for_finetuning(model_to_finetune)
     if not path_to_archive:
         return None
 
+    rasa.shared.utils.cli.print_info(
+        f"Loading Core model from {path_to_archive} for finetuning...",
+    )
+
     with model.unpack_model(path_to_archive) as unpacked:
+        new_fingerprint = await model.model_fingerprint(file_importer)
+        old_fingerprint = model.fingerprint_from_path(unpacked)
+        if not model.can_finetune(old_fingerprint, new_fingerprint, core=True):
+            rasa.shared.utils.cli.print_error_and_exit(
+                "NLU model can not be finetuned."
```

Comment:
```suggestion
                "Core model cannot be finetuned."
```

Reply:
I'll leave this change out of this PR as the change comes from https://github.com/RasaHQ/rasa/pull/7504/files
cc @joejuzl